app/revenue/management/commands/kudos_revenue.py

```diff
@@ -0,0 +1,139 @@
+'''
+    Copyright (C) 2019 Gitcoin Core
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+'''
+
+import logging
+import warnings
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+import requests
+from revenue.models import DigitalGoodPurchase
+
+warnings.filterwarnings("ignore", category=DeprecationWarning)
+logging.getLogger("requests").setLevel(logging.WARNING)
+logger = logging.getLogger(__name__)
+
+REVENUE_ACCOUNT_ADDRESS = '0xdb282cee382244e05dd226c8809d2405b76fbdc9'
+DEFAULT_START_BLOCK = 0
+
+
+def call_etherscan_api(network, params):
+    try:
+        api_url = f'http://api{f"-{network}" if network != "mainnet" else ""}.etherscan.io/api'
+        rs = requests.get(api_url, params).json()
+        records = rs['result']
+        if rs['message'] != 'OK':
+            if len(records) == 0:
+                logger.info(f"No {params['action']} found in blocks {params['startblock']} to {params['endblock']}")
+            else:
+                logger.error(f"* Error during API call => {rs['message']}")
+            return
+        print(f"Total {params['action']} found in blocks {params['startblock']} to {params['endblock']}: {len(records)}")
+        for rec in records:
```

Comment:
seems like the records should be turned from this method and save_incoming_tx() should be called outside of this scope..

just to keep each function small + cohesive

Reply:
my bad! I Just Fixed it!